### PR TITLE
Crawl smart contract addresses from transaction receipts

### PR DESCRIPTION
**Pull request by zomglings**
## Changes

This PR adds a crawler (`ethcrawler contracts update`) which checks for newly deployed smart contracts and adds their addresses to a smart contract registry (`ethereum_smart_contracts` table).

## How to test these changes?

I tested these changes manually against the development database (on Lightsail) and a running Ethereum node.

## Related issues

Resolves https://github.com/bugout-dev/moonstream/issues/25

### CODE REVIEW

File: crawlers/moonstreamcrawlers/ethereum.py [hunk 1]
```diff
@@ -105,3 +110,49 @@ def crawl(block_numbers_list: List[int], with_transactions: bool = False) -> Non
...
+    """
+    web3_client = connect()
+    results: List[Tuple[str, str]] = []
+    with yield_db_session_ctx() as db_session:
+        current_offset = 0
+        limit = 10
+        transactions_remaining = True
+        while transactions_remaining:
+            contract_deployments = (
+                db_session.query(EthereumTransaction)
```
Comment (zomglings) [inline crawlers/moonstreamcrawlers/ethereum.py:133]: This query should only pick transactions which aren't already represented in the `ethereum_smart_contracts` table.

File: db/moonstreamdb/models.py [hunk 2]
```diff
@@ -103,6 +104,18 @@ class EthereumTransaction(Base):  # type: ignore
...
 
 
+class EthereumSmartContract(Base):  # type: ignore
+    __tablename__ = "ethereum_smart_contracts"
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    transaction_hash = Column(
+        VARCHAR(256),
+        ForeignKey("ethereum_transactions.hash", ondelete="CASCADE"),
+        nullable=False,
```
Comment (kompotkot) [inline db/moonstreamdb/models.py:114]: `index=True`

**Review by kompotkot — COMMENTED** [review]
few questions

File: db/moonstreamdb/models.py (hunk 2 above)
Comment (zomglings) [reply db/moonstreamdb/models.py:110]: Done

**Pull request merged**